### PR TITLE
[Communication - Telephony Service] Trunk status API

### PR DESCRIPTION
**Pull request by dzeliar**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [x] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

fix https://github.com/Azure/azure-rest-api-specs/issues/19376
fix https://github.com/Azure/azure-rest-api-specs/issues/19376

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
@dzeliar 
Please see https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship

**Comment by dzeliar** [comment]
As I'll be OOF for the next two weeks, @nikuklic will address any comments

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (jiriburant) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:28]: The paging was not fixed properly, cannot generate code from this specification. Consider adding:
"itemName": "values"

Comment (jiriburant) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:28]: The paging was not fixed properly, cannot generate code from this specification. Consider adding:
"itemName": "values"

File: specification/communication/data-plane/TrunkStatus/readme.md
Comment (jiriburant) [inline specification/communication/data-plane/TrunkStatus/readme.md:54]: Add link to typescript, not csharp.

Comment (jiriburant) [inline specification/communication/data-plane/TrunkStatus/readme.md:54]: Add link to typescript, not csharp.

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (dzeliar) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json]: Removed pageable completely as we expect to have on average just a few records, max is 250

File: specification/communication/data-plane/TrunkStatus/readme.md
Comment (dzeliar) [reply specification/communication/data-plane/TrunkStatus/readme.md]: Updated

**Comment by weidongxu-microsoft** [comment]
There is no update to https://github.com/Azure/azure-rest-api-specs/issues/19349

Let reviewer know after meeting is scheduled and discussion is added (usually Mark's reply in issue).

**Comment by dzeliar** [comment]
Hello @weidongxu-microsoft , our review has been scheduled for today (Wed 29/06/2022)

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:11]: Overall, this new API seems a bit too narrow to me, especially in the following context:

`/sip` already has a GET method. The response to that includes a map of `trunks`. The map there contains trunk FQDN (as the key of the map) and the `sipSignalingPort`.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/SipRouting/preview/2021-05-01-preview/communicationservicessiprouting.json

Now a part of the information about the trunk is split to this separate API under `sip/trunkHealth`.

Two desires:
* Keep information about trunks in one place instead of two.
* Avoid a narrow api path that only stores health information for trunks.

There are a few options to meet these desires:
* Do not introduce the `trunkHealths` collection, instead, expand the `trunks` map under `/sip` and depend on JSON merge-path etc.
  * Should only do this if the set of `trunkHealths` changes a lot, and/or is limited in size.
* Introduce `/sip/trunks` and move the FQDN and `sipSignalingPort` in there. Drop the map under `/sip`
* Introduce `/sip/trunks` to contain static information (like `sipSignalingPort` and `fqdn`) and `sip/trunks/<fqdn>/health` to get the dynamic health for a trunk. The problem here is that it's harder to get the health for *all* trunks.

There is the extra complication that health information is changed by the server, while `sipSignalingPort` is changed by the client (?). This could be confusing. The last option (and the current proposal) mitigates this by keeping the two separate.

--

In any case, splitting the trunk information across a map and a collection seems weird because maps and collections fit different requirements:
e.g, collections must support pagination, with the assumption that collection can grow to be large. In this case, having related information in a map becomes a problem.

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:32]: Avoid including type in the model name (`ListOf`).

You can follow chat's example and call this `TrunkHealthCollection`, or simply `TrunkHealths` (pluralized)

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/Chat/stable/2021-09-07/communicationserviceschat.json#L1238

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:102]: This should be `maxItems`?
https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-01#section-6.4.1

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:100]: The guidelines are fairly clear that we should paginate all collections.

I do see below that we're stating the size of this collection is limited to 250. How hard is this limit? If it's set by laws of physics / a fairly stable standard, we don't need to paginate. If it's based more on current requirements, we should paginate.

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:134]: Food for thought: Should this somehow include _when_ the certificate is expiring?

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:143]: grammar nit: s/send/sent/

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:162]: I'm somewhat uncomfortable with enums that have combinatorial variants: `*PingsAndCalls`. How do we expect this enum to grow in the future? Do we risk blowing up the number of variants? Are clients likely to always have check for multiple related variants in their code (e.g. `if (overallStatus == 'InactiveNoRecentPings' || overallStatus == 'InactiveNoRecentCalls' || overallStatus == 'InactiveNoRecentPingsAndCalls'`)

This smells like it could be split into two separate fields: The status is limited to `NoData | Active | Inactive`. And then for inactive trunks, there are optional reasons `inactiveReasons = [](NoRecentCalls | NoRecentPings)`

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:149]: Food for thought: What is the difference between `NoData` and `PingExpired` from Contoso POV? Can they trigger a ping in the latter case? Otherwise, is there anything different they would do for these two cases? Can we get away with just `NoData` (no reason to tell them _why_ there is no data)

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (prprabhu-ms) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:146]: `NoData` is an unconventional default value for an enum.

A quick search suggests that `unknown` is a more conventional choice (also in my experience):

https://github.com/Azure/azure-rest-api-specs/search?q=Unknown
https://github.com/Azure/azure-rest-api-specs/search?q=Undetermined
https://github.com/Azure/azure-rest-api-specs/search?q=NoData

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (mbakalov) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:11]: FYI related suggestion from a previous review of this api: #19376

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (DominikMe) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:160]: Ideally we stick to lower camelCase and not PascalCase for everything including enum values.

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (mbakalov) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json]: Or maybe `/trunkHealthStatuses` and `/trunkHealthStatuses/{fqdn}`?

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (mbakalov) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:11]: I recall there was also a suggestion to use `expand`: `/sip/trunks?expand=healthStatus` or `/sip?expand=healthStatus` - did we consider this?

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (DominikMe) [inline specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:4]: This service seems fairly minimal in scope, why didn't we integrate it with the SipRouting service? The overhead of another service and SDK isn't small and I would assume as a developer I can't use TrunkHealth if I don't also use SipRounting?

Comment (DominikMe) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json:4]: Could the health data inlined with the rest of the Trunk properties in the SipRouting service?

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (dzeliar) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json]: As agreed on internal review, we will expand `trunks` map under `/sip?expand=trunkHealth`

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (dzeliar) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json]: We can expand heath with multiple properties in future, this is not requested in the current spec

File: specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json
Comment (dzeliar) [reply specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json]: PingError is returned when ping status is not one of the above. Not NoData, OK or Expired. NoData (we don't have any records) is different compared to PingExpired (we have old data but unable to get up-to-date status)

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json
Comment (kdruzycki) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:397]: Maybe you could consider naming this `$expand` instead of `expand`?
That would be in tact with OData standard https://docs.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/using-select-expand-and-value
Also it would be congruent with other usages of `$expand` in this repository https://github.com/Azure/azure-rest-api-specs/search?q=%22%5C%22name%5C%22%3A+%5C%22%24expand%5C%22%22&type=code

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json
Comment (kdruzycki) [inline specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:36]: Error code responses 415, 422, 500 have been removed compared to the previous version...
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/SipRouting/preview/2021-05-01-preview/communicationservicessiprouting.json#L80
Is there a reason for that?

**Review by kdruzycki — CHANGES_REQUESTED** [review]

File: specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json
Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/preview/2022-09-01-preview/communicationservicessiprouting.json:397]: Yes those should have never been there

**Review by kdruzycki — APPROVED** [review]

**Comment by dzeliar** [comment]
Hello everyone,


on yesterdays board review of SipRouting - Trunk health, [Issue #19376](https://github.com/Azure/azure-rest-api-specs/issues/19376), we agreed on the following steps:

1, Rename SipConfigurationExpanded & TrunkExpanded -> ExpandedSipConfiguration & ExpandedTrunk
2, Unify naming Trunk vs. SBC vs. SessionBorderController. After review with our PM we unified naming and now we use only Trunk.

All changes we pushed and the PR is now ready to merge. In order to do so I need an approval from API Stewardship Board.
Please, @mikekistler , @JeffreyRichter, @johanste approve the PR so I can proceed with merge.
Thank you

File: specification/communication/data-plane/SipRouting/preview/2023-01-01-preview/examples/GetSipConfigurationWithExpand.json
Comment (stewartadam) [inline specification/communication/data-plane/SipRouting/preview/2023-01-01-preview/examples/GetSipConfigurationWithExpand.json:4]: I think expand is missing here?

Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/preview/2023-01-01-preview/examples/GetSipConfigurationWithExpand.json]: Yes, thanks you. Added it there

**Comment by dzeliar** [comment]
Hello everyone,

on yesterdays board review of SipRouting - Trunk health, https://github.com/Azure/azure-rest-api-specs/issues/19376, we agreed on the following steps:

1.  You should not create new schemas for Expanded SipConfiguration or ExpandedTrunk. Just add "health" as an optional property the the existing Trunk schema. You can mark health as readOnly: true for documentation purposes.
2.  Don’t use the “$” prefix character on $expand
3.  Enum value for expand should be "trunks/health"
4. Health should be a named schema that is $ref'ed from here. And any objects nested within Health should be described by separate named schemas -- not nested inline.

[OpenAPI diff](https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL2dpdGh1Yi5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvYmxvYi81NzUyMDEzOGNlNTk1YzhhZGI3YTA1OTBkZGU1YzJmNTc1MDJjNzA1L3NwZWNpZmljYXRpb24vY29tbXVuaWNhdGlvbi9kYXRhLXBsYW5lL1NpcFJvdXRpbmcvcHJldmlldy8yMDIyLTEwLTAxLXByZXZpZXcvY29tbXVuaWNhdGlvbnNlcnZpY2Vzc2lwcm91dGluZy5qc29uIiwidGFyZ2V0VXJsIjoiaHR0cHM6Ly9naXRodWIuY29tL0F6dXJlL2F6dXJlLXJlc3QtYXBpLXNwZWNzL2Jsb2IvNTc1MjAxMzhjZTU5NWM4YWRiN2EwNTkwZGRlNWMyZjU3NTAyYzcwNS9zcGVjaWZpY2F0aW9uL2NvbW11bmljYXRpb24vZGF0YS1wbGFuZS9TaXBSb3V0aW5nL3ByZXZpZXcvMjAyMy0wMS0wMS1wcmV2aWV3L2NvbW11bmljYXRpb25zZXJ2aWNlc3NpcHJvdXRpbmcuanNvbiJ9) 

I addressed all of the suggestions above. The changes were pushed, now I need an approval from API Stewardship Board.
Please, @mikekistler , @JeffreyRichter, @DominikMe review and approve the PR so it can be merged.
Thank you

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I flagged a few minor issues.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM.

Same minor comment about redundant "type: object" on "$ref"

File: specification/communication/data-plane/SipRouting/preview/2023-01-01-preview/communicationservicessiprouting.json
Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/preview/2023-01-01-preview/communicationservicessiprouting.json]: Updated

**Comment by dzeliar** [comment]
Thank you for reviews @mikekistler , @weidongxu-microsoft . As merging the PR is restricted to authorized users only, can any of the code-owners perform merge? cc @DominikMe , @scgbear , @JeffreyRichter

**Comment by mikekistler** [comment]
@lirenhe Could you please merge this PR.

**Pull request merged**
